Framework/NexusGeometry/src/H5ForwardCompatibility.cpp

```diff
@@ -0,0 +1,73 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#include "MantidNexusGeometry/H5ForwardCompatibility.h"
+#include <cstring> //memset.
+#include <string>
+
+namespace Mantid {
+namespace NexusGeometry {
+
+namespace H5ForwardCompatibility {
+
+/*
+ * Function "getObjName"
+ * Ported from newer versions of the HDF API for forward compatibility to
+ * support HDF5 version < 1.8.13
+ * https://portal.hdfgroup.org/display/HDF5/Software+Changes+from+Release+to+Release+for+HDF5-1.8
+ */
+
+ssize_t getObjName(const H5::H5Object &obj, char *obj_name, size_t buf_size) {
+  // H5Iget_name will get buf_size-1 chars of the name to null terminate it
+  ssize_t name_size = H5Iget_name(obj.getId(), obj_name, buf_size);
+
+  // If H5Iget_name returns a negative value, raise an exception
+  if (name_size < 0) {
+    throw H5::Exception("getObjName", "H5Iget_name failed");
+  } else if (name_size == 0) {
+    throw H5::Exception("getObjName",
+                        "Object must have a name, but name length is 0");
+  }
+  // Return length of the name
+  return (name_size);
+}
+
+std::string getObjName(const H5::H5Object &obj) {
+  std::string obj_name(""); // object name to return
+
+  // Preliminary call to get the size of the object name
+  ssize_t name_size = H5Iget_name(obj.getId(), NULL, static_cast<size_t>(0));
```

Comment:
`nullptr` is always preferred to `NULL` for type safety.

Reply:
done